upgrade/util/util_test.go

```diff
@@ -46,3 +63,114 @@ func TestProgressMonitorInc(t *testing.T) {
 		t.Fatalf(`targetValue = %v, expectedTargetValue = %v`, targetValue, expectedTargetValue)
 	}
 }
+
+func TestCheckUpgradePathSupported(t *testing.T) {
+	testCases := []struct {
+		name           string
+		currentVersion string
+		upgradeVersion string
+		expectError    bool
+	}{
+		{
+			name:           "new install",
+			currentVersion: "",
+			upgradeVersion: "v1.5.0",
+			expectError:    false,
+		},
+		{
+			name:           "invalid upgrade version",
+			currentVersion: "v1.5.0",
+			upgradeVersion: "v1.5.x",
+			expectError:    true,
```

Comment:
This should be supported. Or, what example data you should use is `1.5.0` -> `1.5.1`. You can also add one more `1.5.1` -> `1.5.2`

1.5.0/1.4.x can be upgraded to 1.5.0

Reply:
Modified the misleading version string from `v1.5.x` to `v1.5.error`.
And added test cases from `v1.5.0` to `v1.5.1` and `v1.5.2` to `v1.5.3`

Reply:
As discussed, 1.5.x is used for documentation so it's not a valid case. We don't need to test it.

Reply:
Removed invalid version test case, and thanks for explaining.